soroban-env-host/src/budget.rs

```diff
@@ -703,36 +1056,53 @@ impl Budget {
         })
     }
 
+    #[cfg(test)]
+    pub(crate) fn override_model_with_scaled_params(
+        &self,
+        ty: ContractCostType,
+        const_cpu: u64,
+        lin_cpu: ScaledU64,
+        const_mem: u64,
+        lin_mem: ScaledU64,
+    ) -> Result<(), HostError> {
+        let mut bgt = self.0.try_borrow_mut_or_err()?;
+
+        let cpu_model = bgt.cpu_insns.get_cost_model_mut(ty);
+        cpu_model.const_term = const_cpu;
+        cpu_model.lin_term = lin_cpu;
+
+        let mem_model = bgt.mem_bytes.get_cost_model_mut(ty);
+        mem_model.const_term = const_mem;
+        mem_model.lin_term = lin_mem;
+        Ok(())
+    }
+
+    #[cfg(test)]
+    pub(crate) fn override_model_with_unscaled_params(
+        &self,
+        ty: ContractCostType,
+        const_cpu: u64,
+        lin_cpu: u64,
+        const_mem: u64,
+        lin_mem: u64,
+    ) -> Result<(), HostError> {
+        self.override_model_with_scaled_params(
+            ty,
+            const_cpu,
+            ScaledU64::from_unscaled_u64(lin_cpu),
+            const_mem,
+            ScaledU64::from_unscaled_u64(lin_mem),
+        )
+    }
+
     #[cfg(any(test, feature = "testutils"))]
     pub fn reset_fuel_config(&self) -> Result<(), HostError> {
         self.0.try_borrow_mut_or_err()?.fuel_config.reset();
         Ok(())
     }
 
-    pub(crate) fn get_cpu_insns_remaining_as_fuel(&self) -> Result<u64, HostError> {
-        let cpu_remaining = self.get_cpu_insns_remaining()?;
-        let cpu_per_fuel = self
-            .0
-            .try_borrow_or_err()?
-            .cpu_insns
-            .get_cost_model(ContractCostType::WasmInsnExec)
-            .linear_term;
```

Comment:
This is the fuel conversion bug I mentioned. This should have be `const_term` instead of `linear_term`. It is basically undercharging cpu for wasm instructions by a factor of 7.

Reply:
Interesting! Why wasn't this noticed numerically?

Reply:
I'm glad you asked! My above statement about undercharging wasn't accurate.
This code is called when we calculate how much fuel (converted from the cpu budget) should be provide to the VM. So it affects the "fuel limit" perceived by the VM. The bug here means the VM will think it has a fuel budget that's 7x larger than what it is.
When the VM execution comes back, we charge the number of fuel units back to the cpu budget, which goes through the normal model evaluation flow, with the correct model, and the amount of cpu consumed and charged to the budget is correct. 
So the implication of this bug is: it only affects over-the-cpu-limit contracts, where they will be allowed to 7x run longer before trapping. All contracts will be charged the correct amount, the violating ones might be charged 7x the budget limit. 
I have enhanced the `test_vm_fuel_metering` test to cover this scenario.